utils/rsnapreport.pl

```diff
@@ -72,18 +77,33 @@ ($)
 }
 
 while (my $line = nextLine(\@rsnapout)){
-	if($line =~ /^[\/\w]+\/rsync/) { # find start rsync command line
+        while($line =~ /\s+\\$/){ # combine wrapped lines
+		$line =~ s/\\$//g;
+		$line .= nextLine(\@rsnapout);
+	}
+	if($line =~ /^[\/\w]+\/lvcreate\h+-[-\w]/) { # Look for LVM snapshot
+		# Extract the LVM logical volume from the lvcreate command.
+		my $lvpath = (split /\s+/, $line)[-1];
+		my ($vg, $lv) = (File::Spec->splitdir($lvpath))[-2,-1];
+		$linux_lvm_lv = 'lvm://' . $vg . '/' . $lv . '/';
+	}
+	# find start rsync command line
+	elsif($line =~ /^[\/\w]+\/rsync\h+-[-\w]/) {
 		my @rsynccmd=();
-		while($line =~ /\s+\\$/){ # combine wrapped lines
-			$line =~ s/\\$//g;
-			$line .= nextLine(\@rsnapout);
-		}
 		push(@rsynccmd,split(/\s+/,$line)); # split into command components
-		my $source = $rsynccmd[-2]; # count backwards: source always second to last
+		my $source;
+		# Use LVM logical volume name if it exists.
+		if ($linux_lvm_lv) {
+			$source = $linux_lvm_lv;
+		}
+		else {
+			# count backwards: source always second to last
```

Comment:
At a glance- is it possible to guarantee this placement? If so, it would be nice to add some test cases to ensure that always remains true. It might be nice to add another test case or two to help avoid future regressions.

Reply:
Sorry, it has been a long time since I worked on this. Can you be more specific? Guarantee placement of what? Also, are there any existing test cases for rsnapreport?

Reply:
Indeed, the project has been silent for a long time now.

```
# count backwards: source always second to last
$source = $rsynccmd[-2];
```

I haven't taken a very deep dive into the source yet, but I'm curious if we can guaranteed that "source" is *always* second to last. My concern is that this isn't a requirement of rsync and something swapped in the code could potentially cause a problem here.

Reply:
The short answer is that I don't have time to add a test case.

The longer answer is that the dependence on position was present in the original rsnapreport code, so this isn't something new. While changes to rsnapshot could potentially reorder the rsync command line, this seems rather unlikely to me. In any event, only rsnapreport would be broken by this. Finally, if there were existing test cases for rsnapreport, I would be more inclined to take this on, but creating an rsnapshot test case from scratch is not something I have time for.

Reply:
The lack of time is understandable. I still haven't dug through the source a whole lot, but it sounds like you have more than enough reason to be sure that consistency will be there.

I'm still trying to wrap my head around testing stuff. I'll look into making some test cases for rsnapreport. (I don't have access to merge anything; just trying to help out development)

Thanks for taking the time to respond.